### PR TITLE
fix(telephony): access to configuration menu

### PR DESCRIPTION
**Pull request by radireddy**
ref: #MANAGER-7165

Signed-off-by: Ravindra Adireddy <ravindra.adireddy@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-7165
| License          | BSD 3-Clause

## Description

Telecom SVI enhancements

## Related

- [x] TRANS-37485

### CODE REVIEW

**Review by antleblanc — COMMENTED** [review]
According to the commit message, I'm a bit confused.
Is it a `feat` or a `fix`?

**Comment by radireddy** [comment]
> According to the commit message, I'm a bit confused.
> Is it a `feat` or a `fix`?

jira ticket is of type task. I think its enhancements.

File: packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/extension/rule/rule.controller.js [hunk 1]
```diff
@@ -4,13 +4,15 @@ import indexOf from 'lodash/indexOf';
...
   constructor(
+    $state,
     $q,
     $translate,
     atInternet,
     autoScrollOnToggle,
     TelephonyMediator,
     TucToast,
   ) {
+    this.$state = $state;
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/extension/rule/rule.controller.js:15]: Can't be injected through resolve ?

File: packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/extension/rule/rule.html [hunk 2]
```diff
@@ -142,6 +142,19 @@
                 </li>
                 <!-- EDIT ACTION -->
                 <li class="divider"></li>
+
+                <!-- EDIT interactif menu -->
+                <li role="menuitem" data-ng-if="$ctrl.rule.action === 'ivr'">
+                    <button
+                        type="button"
+                        class="btn btn-link w-100"
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/extension/rule/rule.html:150]: Not a ui-kit class ?

File: packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/extension/rule/rule.controller.js (hunk 1 above)
Comment (radireddy) [reply packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/ovhPabx/dialplan/extension/rule/rule.controller.js:15]: very difficult and lot of refactoring due to legacy code.

File: packages/manager/apps/telecom/src/app/telecom/telephony/alias/configuration/ovhPabx/sounds/sounds.controller.js [hunk 3]
```diff
@@ -67,5 +67,19 @@ export default /* @ngInject */ function TelecomTelephonyAliasConfigurationOvhPab
...
+      switch (self.number.feature.featureType) {
+        case 'cloudIvr':
+          return 'SVI';
+        case 'contactCenterSolutionExpert':
+          return 'CCS';
+        default:
+          return null;
+      }
+    }
+    return null;
```
Comment (mohammed-zahaf) [inline packages/manager/apps/telecom/src/app/telecom/telephony/alias/configuration/ovhPabx/sounds/sounds.controller.js:81]: i think we can have something like:
```suggestion
    const features = {'cloudIvr': 'SVI', 'contactCenterSolutionExpert': 'CCS'}
    return features[self.number.feature.featureType] || null;
```

**Review by ganeshkumar1989 — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/telephony/alias/configuration/ovhPabx/sounds/sounds.controller.js [hunk 4]
```diff
@@ -67,5 +69,19 @@ export default /* @ngInject */ function TelecomTelephonyAliasConfigurationOvhPab
...
+  self.getFeatureType = function $onInit() {
+    if (self.number.feature?.featureType) {
+      switch (self.number.feature.featureType) {
+        case 'cloudIvr':
+          return FEATURES.cloudIvr;
+        case 'contactCenterSolutionExpert':
+          return FEATURES.contactCenterSolutionExpert;
+        default:
+          return null;
+      }
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/telephony/alias/configuration/ovhPabx/sounds/sounds.controller.js:81]:
```suggestion
      return FEATURES[self.number.feature.featureType] 
```
?

**Review by antleblanc — APPROVED** [review]

**Pull request merged**